### ISSUE TITLE
Missing @DatabaseContext parameter in sp_DatabaseRestore

### ISSUE THREAD

**Issue by mKarous** [BrentOzarULTD/SQL-Server-First-Responder-Kit#2269]
The sp_DatabaseRestore stored procedure uses the CommandExecute procedure from the Ola Hallengren solution, but it is no more compatible with the latest version of the Ola Hallengren solution. The latest version of the CommandExecute procedure (2020-01-14) requires new @DatabaseContext input parameter, but this parameter is missing in the current version of the sp_DatabaseRestore procedure and so the procedure sp_DatabaseRestore fails.

I hotfixed this in my local installations by adding this parameter into all calls of CommandExecute procedure inside sp_DatabaseRestore with the constant value of @DatabaseContext = 'master' which fixes the issue for me.

**Comment by BrentOzar** [comment]
Ouch - totally blows that there's a new parameter w/o a default value. In order to fix this, we would need to examine the proc's parameters and then add that new parameter on the fly based on the version of CommandExecute. Is there a reason that CommandExecute shouldn't just have a default value instead?

**Comment by estraffen** [comment]
Seeing the same issue here. Would it make sense to completely remove the dependency on the Ola Hallengren scripts? They look to only be used in sp_DatabaseRestore, specifically the stored procedures dbo.CommandExecute and dbo.DatabaseIntegrityCheck. These could seemingly be replaced with DBCC CHECKDB and sp_executesql calls instead. That would remove the dependency and hopefully prevent future breaks.

**Comment by BrentOzar** [comment]
Sure, that would work too - is there anybody here who wants to code that? (I don't use this proc anymore, and there are already a couple of pull requests waiting for testing.)

**Comment by estraffen** [comment]
I don't mind updating it, so you can assign to me.

**Comment by BrentOzar** [comment]
OK, cool, thanks!

**Comment by BrentOzar** [comment]
Looks good! I merged this w/o testing it, which is unusual, but I don't use this proc a lot, and it looks fine at first glance. Nice work! It's in the dev branch now, and it'll be in the February release.

**Comment by estraffen** [comment]
Thanks and awesome to hear!

**Comment by RobAshton-IM** [comment]
@estraffen, @BrentOzar, I wish I'd seen this bug sooner :(

Yes, Ola Hallengren has changed the input parameters to **CommandExecute** in a breaking way, but he's done so for a good reason within his project. The new _DatabaseContext_ parameter is mandatory as it's the name of the database that your command should be executed within instead of it being assumed to be the same as the _DatabaseName_ parameter.

By changing sp_DatabaseRestore so it no longer uses CommandExecute, we've lost the fringe benefits we get around command logging from CommandExecute. I don't know about other DBA's out there, but I have auditing and monitoring in place that watches the CommandLog table to get presence/absence of events and their timings to get alerts out when things start moving away from nominal, so I've now got a hole in my processes that I need to patch.

To give an example of what I think should be done here, a database restore command should be called from within the master database, so the right fix for that sort of command should have been what @mKarous did and add this new parameter set to "master" in the **CommandExecute** call.

I've created bug #2306 to track an alternative fix for this issue.
